### PR TITLE
Kubevirt: add basic actions to VM list

### PR DESCRIPTION
**Pull request by suomiy**
depends on:

- [x] #1758 
- [x] #1829

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+const isImporting = (vm, { pods, migrations }: ActionArgs) => {
+  const status = getVmStatus(vm, pods, migrations);
+  return (
+    status && [VM_STATUS_IMPORTING, VM_STATUS_V2V_CONVERSION_IN_PROGRESS].includes(status.status)
+  );
+};
+
+const menuActionStart = (kind, vm, actionArgs) => {
```
Comment (spadgett) [inline frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx]: You'll want to add RBAC checks to these actions. For example,

https://github.com/openshift/console/blob/master/frontend/public/components/utils/kebab.tsx#L64-L78

File: frontend/packages/kubevirt-plugin/src/components/modals/start-stop-vm-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+} from '@console/internal/components/factory';
+
+import { k8sPatch } from '@console/internal/module/k8s';
+
+import { getPxeBootPatch } from 'kubevirt-web-ui-components';
+import { V1VirtualMachine } from 'kubevirt-typescript-api/esm';
+import { getName, getNamespace } from '@console/shared';
+import { VirtualMachineModel } from '../../models';
+
+class StartStopVmModal extends PromiseComponent<StartStopVmModalProps, PromiseComponentState> {
```
Comment (spadgett) [inline frontend/packages/kubevirt-plugin/src/components/modals/start-stop-vm-modal.tsx]: Prefer functional components for new components

File: frontend/packages/kubevirt-plugin/src/components/modals/start-stop-vm-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  render() {
+    const { vm } = this.props;
+    const action = this.props.start ? 'Start' : 'Stop';
+    return (
+      <form onSubmit={this.submit} name="form" className="modal-content">
+        <ModalTitle>{action} Virtual Machine</ModalTitle>
+        <ModalBody>
+          Are you sure you want to {action.toLowerCase()} <strong>{getName(vm)}</strong>
+          <span>
```
Comment (spadgett) [inline frontend/packages/kubevirt-plugin/src/components/modals/start-stop-vm-modal.tsx]: Unnecessary span?

File: frontend/packages/kubevirt-plugin/src/components/modals/start-stop-vm-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      </form>
+    );
+  }
+}
+
+type StartStopVmModalProps = {
+  vm: V1VirtualMachine;
+  start: boolean;
+  cancel: (e?: Event) => void;
+  close: (e?: Event) => void;
```
Comment (spadgett) [inline frontend/packages/kubevirt-plugin/src/components/modals/start-stop-vm-modal.tsx]: I don't see us passing the event to these callbacks in the implementation. We should take it out if we're not using it.

File: frontend/packages/kubevirt-plugin/src/selectors/vm/index.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import get from 'ts-get';
+
+import { V1VirtualMachine } from 'kubevirt-typescript-api/esm';
+
+export const isVmRunning = (value: V1VirtualMachine) => get(value, (v) => v.spec.running, false);
```
Comment (spadgett) [inline frontend/packages/kubevirt-plugin/src/selectors/vm/index.ts]: `isVMRunning` etc

**Comment by suomiy** [comment]
addresses all the comments

File: frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import {
+  getVmStatus,
+  VM_STATUS_IMPORTING,
+  VM_STATUS_V2V_CONVERSION_IN_PROGRESS,
+} from 'kubevirt-web-ui-components';
+
+import { asAccessReview, Kebab } from '@console/internal/components/utils';
+import { isVMRunning } from '../../selectors/vm';
+import { startStopVmModal } from '../modals/start-stop-vm-modal';
+import { K8sKind, PodKind } from '../../../../../public/module/k8s';
```
Comment (jtomasek) [inline frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx]:
```suggestion
import { K8sKind, PodKind } from '@console/internal/module/k8s';
```

File: frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx [hunk 7]
```diff
@@ -1,101 +1,125 @@
...
 import {
+  VirtualMachineInstanceMigrationModel,
   VirtualMachineModel,
   // VirtualMachineInstanceModel,
-  // VirtualMachineInstanceMigrationModel,
 } from '../../models';
 
+import { VMKind } from '../../types';
+import { menuActions } from './menu-actions';
+import { PodKind } from '../../../../../public/module/k8s';
```
Comment (jtomasek) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm.tsx]:
```suggestion
import { PodKind } from '@console/internal/module/k8s';
```

File: frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx (hunk 6 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx]: Thanks, fixed.

File: frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import { isVMRunning } from '../../selectors/vm';
+import { startStopVmModal } from '../modals/start-stop-vm-modal';
+import { VMKind } from '../../types/vm';
+
+type ActionArgs = {
+  pods: PodKind[];
+  migrations: any[];
+};
+
+const isImporting = (vm, { pods, migrations }: ActionArgs) => {
```
Comment (rawagner) [inline frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx]:
```suggestion
const isImporting = (vm: VMKind, { pods, migrations }: ActionArgs) => {
```

**Comment by suomiy** [comment]
fixed

**Comment by suomiy** [comment]
added types and new PR dependency

**Pull request merged**